### PR TITLE
added continuous flag option

### PR DESCRIPTION
**Pull request by arlemi**

### CODE REVIEW

File: services/speech_to_text/v1.js [hunk 1]
```diff
@@ -63,14 +63,20 @@ module.exports = function (RED) {
       }
 
       if (!config.lang) {
-        var message = 'Missing audio language configuration, unable to process speech.';
-        node.error(message, msg)
```
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_lang' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2389271691/issues/source?bid=3225115&fileBranchId=3229003#l66)

File: services/speech_to_text/v1.js [hunk 2]
```diff
@@ -63,14 +63,20 @@ module.exports = function (RED) {
...
+        var message_err_lang = 'Missing audio language configuration, unable to process speech.';
+        node.error(message_err_lang, msg)
         return;
       }
 
       if (!config.band) {
-        var message = 'Missing audio band configuration, unable to process speech.';
-        node.error(message, msg)
+        var message_err_band = 'Missing audio band configuration, unable to process speech.';
```
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_band' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2389271691/issues/source?bid=3225115&fileBranchId=3229003#l73)

File: services/speech_to_text/v1.js [hunk 3]
```diff
@@ -42,17 +42,17 @@ module.exports = function (RED) {
...
         return;
       }
 
       username = username || this.credentials.username;
       password = password || this.credentials.password;
 
       if (!username || !password) {
-        var message = 'Missing Speech To Text service credentials';
-        node.error(message, msg)
+        var message_err_credentials = 'Missing Speech To Text service credentials';
```
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_credentials' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2389271691/issues/source?bid=3225115&fileBranchId=3229003#l55)

File: services/speech_to_text/v1.js [hunk 4]
```diff
@@ -63,14 +63,20 @@ module.exports = function (RED) {
...
-        var message = 'Missing audio language configuration, unable to process speech.';
-        node.error(message, msg)
+        var message_err_lang = 'Missing audio language configuration, unable to process speech.';
+        node.error(message_err_lang, msg)
         return;
       }
 
       if (!config.band) {
-        var message = 'Missing audio band configuration, unable to process speech.';
-        node.error(message, msg)
```
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_band' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2389271691/issues/source?bid=3225115&fileBranchId=3229003#l72)

File: services/speech_to_text/v1.js [hunk 5]
```diff
@@ -63,14 +63,20 @@ module.exports = function (RED) {
       }
 
       if (!config.lang) {
-        var message = 'Missing audio language configuration, unable to process speech.';
-        node.error(message, msg)
+        var message_err_lang = 'Missing audio language configuration, unable to process speech.';
```
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_lang' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2389271691/issues/source?bid=3225115&fileBranchId=3229003#l67)

File: services/speech_to_text/v1.js [hunk 6]
```diff
@@ -63,14 +63,20 @@ module.exports = function (RED) {
...
 
       if (!config.band) {
-        var message = 'Missing audio band configuration, unable to process speech.';
-        node.error(message, msg)
+        var message_err_band = 'Missing audio band configuration, unable to process speech.';
+        node.error(message_err_band, msg)
+        return;
+      }
+
+      if (!config.continuous) {
```
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_continuous' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2389271691/issues/source?bid=3225115&fileBranchId=3229003#l78)

File: services/speech_to_text/v1.js [hunk 7]
```diff
@@ -63,14 +63,20 @@ module.exports = function (RED) {
...
+        var message_err_continuous = 'Missing continuous details, unable to process speech.';
```
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_continuous' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2389271691/issues/source?bid=3225115&fileBranchId=3229003#l79)

File: services/speech_to_text/v1.js [hunk 8]
```diff
@@ -42,17 +42,17 @@ module.exports = function (RED) {
...
+        node.error(message_err_payload, msg)
         return;
       }
 
       username = username || this.credentials.username;
       password = password || this.credentials.password;
 
       if (!username || !password) {
-        var message = 'Missing Speech To Text service credentials';
-        node.error(message, msg)
```
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_credentials' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2389271691/issues/source?bid=3225115&fileBranchId=3229003#l54)

File: services/speech_to_text/v1.js [hunk 9]
```diff
@@ -85,17 +91,18 @@ module.exports = function (RED) {
 
       var s2t = function (audio, format, cb) {
         if (format !== 'wav' && format !== 'flac' && format !== 'ogg') {
-          var message = 'Audio format (' + format + ') not supported, must be encoded as WAV, FLAC or OGG.';
-          node.error(message, msg)
+          var message_err_format = 'Audio format (' + format + ') not supported, must be encoded as WAV, FLAC or OGG.';
```
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_format' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2389271691/issues/source?bid=3225115&fileBranchId=3229003#l95)

File: services/speech_to_text/v1.js [hunk 10]
```diff
@@ -42,17 +42,17 @@ module.exports = function (RED) {
 
     this.on('input', function (msg) {
       if (!msg.payload) {
-        var message = 'Missing property: msg.payload';
-        node.error(message, msg)
+        var message_err_payload = 'Missing property: msg.payload';
+        node.error(message_err_payload, msg)
```
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_payload' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2389271691/issues/source?bid=3225115&fileBranchId=3229003#l46)

File: services/speech_to_text/v1.js (hunk 2 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_band' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391504090/issues/source?bid=3225115&fileBranchId=3229003#l73)

File: services/speech_to_text/v1.js (hunk 1 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_lang' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391504090/issues/source?bid=3225115&fileBranchId=3229003#l66)

File: services/speech_to_text/v1.js (hunk 3 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_credentials' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391504090/issues/source?bid=3225115&fileBranchId=3229003#l55)

File: services/speech_to_text/v1.js (hunk 4 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_band' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391504090/issues/source?bid=3225115&fileBranchId=3229003#l72)

File: services/speech_to_text/v1.js (hunk 5 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_lang' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391504090/issues/source?bid=3225115&fileBranchId=3229003#l67)

File: services/speech_to_text/v1.js (hunk 6 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_continuous' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391504090/issues/source?bid=3225115&fileBranchId=3229003#l78)

File: services/speech_to_text/v1.js (hunk 8 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_credentials' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391504090/issues/source?bid=3225115&fileBranchId=3229003#l54)

File: services/speech_to_text/v1.js (hunk 7 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_continuous' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391504090/issues/source?bid=3225115&fileBranchId=3229003#l79)

File: services/speech_to_text/v1.js (hunk 9 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_format' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391504090/issues/source?bid=3225115&fileBranchId=3229003#l95)

File: services/speech_to_text/v1.js (hunk 10 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_payload' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391504090/issues/source?bid=3225115&fileBranchId=3229003#l46)

File: services/speech_to_text/v1.js [hunk 11]
```diff
@@ -42,17 +42,17 @@ module.exports = function (RED) {
 
     this.on('input', function (msg) {
       if (!msg.payload) {
-        var message = 'Missing property: msg.payload';
-        node.error(message, msg)
+        var message_err_payload = 'Missing property: msg.payload';
```
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_payload' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391511660/issues/source?bid=3225115&fileBranchId=3229003#l45)

File: services/speech_to_text/v1.js (hunk 8 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_credentials' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391511660/issues/source?bid=3225115&fileBranchId=3229003#l54)

File: services/speech_to_text/v1.js (hunk 6 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_continuous' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391511660/issues/source?bid=3225115&fileBranchId=3229003#l78)

File: services/speech_to_text/v1.js (hunk 3 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391511660/issues/source?bid=3225115&fileBranchId=3229003#l55)

File: services/speech_to_text/v1.js (hunk 4 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_band' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391511660/issues/source?bid=3225115&fileBranchId=3229003#l72)

File: services/speech_to_text/v1.js (hunk 10 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391511660/issues/source?bid=3225115&fileBranchId=3229003#l46)

File: services/speech_to_text/v1.js (hunk 7 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391511660/issues/source?bid=3225115&fileBranchId=3229003#l79)

File: services/speech_to_text/v1.js (hunk 5 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391511660/issues/source?bid=3225115&fileBranchId=3229003#l67)

File: services/speech_to_text/v1.js (hunk 9 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391511660/issues/source?bid=3225115&fileBranchId=3229003#l95)

File: services/speech_to_text/v1.js [hunk 12]
```diff
@@ -85,17 +91,18 @@ module.exports = function (RED) {
 
       var s2t = function (audio, format, cb) {
         if (format !== 'wav' && format !== 'flac' && format !== 'ogg') {
-          var message = 'Audio format (' + format + ') not supported, must be encoded as WAV, FLAC or OGG.';
-          node.error(message, msg)
```
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_format' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391511660/issues/source?bid=3225115&fileBranchId=3229003#l94)

File: services/speech_to_text/v1.js (hunk 2 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_band' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391621463/issues/source?bid=3225115&fileBranchId=3229003#l73)

File: services/speech_to_text/v1.js (hunk 1 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_lang' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391621463/issues/source?bid=3225115&fileBranchId=3229003#l66)

File: services/speech_to_text/v1.js (hunk 3 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_credentials' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391621463/issues/source?bid=3225115&fileBranchId=3229003#l55)

File: services/speech_to_text/v1.js (hunk 4 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_band' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391621463/issues/source?bid=3225115&fileBranchId=3229003#l72)

File: services/speech_to_text/v1.js (hunk 5 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_lang' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391621463/issues/source?bid=3225115&fileBranchId=3229003#l67)

File: services/speech_to_text/v1.js (hunk 6 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_continuous' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391621463/issues/source?bid=3225115&fileBranchId=3229003#l78)

File: services/speech_to_text/v1.js (hunk 8 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_credentials' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391621463/issues/source?bid=3225115&fileBranchId=3229003#l54)

File: services/speech_to_text/v1.js (hunk 7 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_continuous' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391621463/issues/source?bid=3225115&fileBranchId=3229003#l79)

File: services/speech_to_text/v1.js (hunk 9 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_format' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391621463/issues/source?bid=3225115&fileBranchId=3229003#l95)

File: services/speech_to_text/v1.js (hunk 10 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js]: Codacy Issue found: [Identifier 'message_err_payload' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391621463/issues/source?bid=3225115&fileBranchId=3229003#l46)

File: services/speech_to_text/v1.js (hunk 11 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js:45]: Codacy Issue found: [Identifier 'message_err_payload' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391825509/issues/source?bid=3225115&fileBranchId=3229003#l45)

File: services/speech_to_text/v1.js (hunk 8 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js:55]: Codacy Issue found: [Identifier 'message_err_credentials' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391825509/issues/source?bid=3225115&fileBranchId=3229003#l54)

File: services/speech_to_text/v1.js (hunk 6 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js:77]: Codacy Issue found: [Identifier 'message_err_continuous' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391825509/issues/source?bid=3225115&fileBranchId=3229003#l78)

File: services/speech_to_text/v1.js (hunk 3 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js:54]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391825509/issues/source?bid=3225115&fileBranchId=3229003#l55)

File: services/speech_to_text/v1.js (hunk 4 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js:73]: Codacy Issue found: [Identifier 'message_err_band' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391825509/issues/source?bid=3225115&fileBranchId=3229003#l72)

File: services/speech_to_text/v1.js (hunk 10 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js:46]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391825509/issues/source?bid=3225115&fileBranchId=3229003#l46)

File: services/speech_to_text/v1.js (hunk 7 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js:78]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391825509/issues/source?bid=3225115&fileBranchId=3229003#l79)

File: services/speech_to_text/v1.js (hunk 5 above)
Comment (yrezgui) [inline services/speech_to_text/v1.js:66]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391825509/issues/source?bid=3225115&fileBranchId=3229003#l67)

File: services/speech_to_text/v1.js [hunk 13]
```diff
@@ -85,17 +91,20 @@ module.exports = function (RED) {
...
+          var message_err_format = 'Audio format (' + format + ') not supported, must be encoded as WAV, FLAC or OGG.';
```
Comment (yrezgui) [inline services/speech_to_text/v1.js:94]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391825509/issues/source?bid=3225115&fileBranchId=3229003#l95)

File: services/speech_to_text/v1.js [hunk 14]
```diff
@@ -85,17 +91,20 @@ module.exports = function (RED) {
...
         }
 
         if (format === 'ogg') format += ';codecs=opus';
 
         var params = {
           audio: audio,
+          content_type: 'audio/' + format,
           model: model,
-          content_type: 'audio/' + format 
+          continuous: true,
```
Comment (yrezgui) [inline services/speech_to_text/v1.js:105]: Codacy Issue found: [Identifier 'inactivity_timeout' is not in camel case.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2391825509/issues/source?bid=3225115&fileBranchId=3229003#l106)

**Comment by kwiatks** [comment]
Tested - requires Default value on Continuous option to be set to No

**Pull request closed without merging**